components/automate-gateway/handler/deployment.go

```diff
@@ -48,3 +48,34 @@ func (s *DeploymentServer) ServiceVersions(ctx context.Context, _ *deployment.Se
 
 	return &deployment.ServiceVersionsResponse{Services: gatewayServices}, nil
 }
+
+func (s *DeploymentServer) GetDeploymentStatus(ctx context.Context, _ *deployment.DeploymentStatusRequest) (*deployment.DeploymentStatusResponse, error) {
+	status, err := s.deploymentClient.Status(ctx, &api.StatusRequest{})
+	if err != nil {
+		logrus.WithError(err).Error("failed to get deployment status")
+		return nil, err
```

Comment:
We might consider using `status.Errorf` here so we can return an appropriate status code.